### PR TITLE
👌 IMPROVE: Error msg for unimplemented directives

### PR DESCRIPTION
**Pull request by tfiers**
Suggest using `{eval-rst}` and link to its documentation.

As suggested in https://github.com/executablebooks/MyST-Parser/issues/284#issuecomment-756434092

### CODE REVIEW

File: myst_parser/mocking.py [hunk 1]
```diff
@@ -235,12 +235,15 @@ def __getattr__(self, name):
...
-            msg = "{cls} has not yet implemented attribute '{name}'".format(
-                cls=type(self).__name__, name=name
+            msg = (
+                f"{cls} has not yet implemented attribute '{name}'. "
+                "You can parse RST directly via the `{eval-rst}` directive: "
+                "https://myst-parser.readthedocs.io/en/latest/using/syntax.html#how-directives-parse-content"  # noqa: E501
             )
-            raise MockingError(msg).with_traceback(sys.exc_info()[2])
-        msg = "{cls} has no attribute {name}".format(cls=type(self).__name__, name=name)
+        else:
```
Comment (choldgraf) [inline myst_parser/mocking.py:245]: Can you add a comment that describes under what condition the "if" statement will return False and thus the simpler error message?

Comment (tfiers) [reply myst_parser/mocking.py:245]: Something like this:
```py
if hasattr(Body, name):
    ...
else:
    # The requested `name` is not a docutils Body element
    # (such as "admoniton", "block_quote", "paragraph", …)
    ...
```
?

Comment (choldgraf) [reply myst_parser/mocking.py:245]: that seems good 👍

Comment (tfiers) [reply myst_parser/mocking.py:245]: I made the code a bit DRY-er but did not introduce the if-else logic, that was already there (i.e. this is a pure refactor -- no different behavior).
Maybe the original author (Chris Sewell) can explain the reasoning behind the two cases.

Comment (tfiers) [reply myst_parser/mocking.py:245]: Anyway I added the above comment in a commit, so should be good to go :)

**Comment by choldgraf** [comment]
I believe this is just changing the error message so I think the codecov fail isn't relevant here. Gonna merge away. Thanks so much @tfiers for the improvement :-)

**Pull request merged**